### ISSUE TITLE
Inet4Address round trip returns 0.0.0.0

### ISSUE THREAD

**Issue by jangoolie** [EsotericSoftware/kryo#511]
```scala
import com.esotericsoftware.kryo._
import java.net.Inet4Address
import io.{ Output, Input }
import java.io._

val kryo = new Kryo()

val output = new Output(new FileOutputStream("file.bin"))
val ip = java.net.InetAddress.getByName("123.123.123.123")
kryo.writeObject(output, ip)
output.close

val input = new Input(new FileInputStream("file.bin"));
val ip2 = kryo.readObject(input, classOf[Inet4Address])

scala> ip
res3: java.net.InetAddress = /123.123.123.123

scala> ip2
res4: java.net.Inet4Address = /0.0.0.0
```

:(

**Comment by gurubook** [comment]
Same problem here, for Inet4Address and Inet6Address

`
package kryo.test;

import static org.junit.Assert.assertTrue;

import java.io.ByteArrayOutputStream;
import java.net.InetAddress;
import java.net.UnknownHostException;

import org.junit.Test;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;

public class KryoSerializerTest {

	public static class Addresses {
		public Addresses() {
			super();	
		}
		
		public Addresses(InetAddress ipv4, InetAddress ipv6) {
			super();
			this.ipv4 = ipv4;
			this.ipv6 = ipv6;
		}
		
		private InetAddress ipv4;
		private InetAddress ipv6;
		
		public InetAddress getIpv4() {
			return ipv4;
		}

		public void setIpv4(InetAddress ipv4) {
			this.ipv4 = ipv4;
		}

		public InetAddress getIpv6() {
			return ipv6;
		}

		public void setIpv6(InetAddress ipv6) {
			this.ipv6 = ipv6;
		}
	}
	
	@Test
	public void inetAddressSerializationTest() throws UnknownHostException {
		Kryo kryo = new Kryo();
		
		// initialize addresses
		Addresses addresses = new Addresses();
		addresses.setIpv4(InetAddress.getByName("192.168.0.1"));
		addresses.setIpv6(InetAddress.getByName("::1"));
		assertTrue(addresses.ipv4.getHostAddress().equals("192.168.0.1"));
		assertTrue(addresses.ipv6.getHostAddress().equals("0:0:0:0:0:0:0:1"));
		
		ByteArrayOutputStream bios = new ByteArrayOutputStream();		
		Output output = new Output(bios);
		kryo.writeObject(output, addresses);		
		output.close();
		
		Input input = new Input(bios.toByteArray());
		Addresses deserializedAddresses = kryo.readObject(input, Addresses.class);
		assertTrue(deserializedAddresses.ipv4.getHostAddress().equals("192.168.0.1"));
		assertTrue(deserializedAddresses.ipv6.getHostAddress().equals("0:0:0:0:0:0:0:1"));
	}
}
`

**Comment by magro** [comment]
There's just no default serializer for these classes. You could write and contribute one, PRs welcome!

**Comment by jangoolie** [comment]
Hey

I've prepared a pull request for these. I've implemented the serialises, added them to the default serializers and added tests for them. I'm now faced with this however:

Failed tests: 
  SerializationCompatTest.testDefaultSerializers:103 The registered default serializers changed.
Because serialization compatibility shall be checked for default serializers, you must extend SerializationCompatTestData.TestData to have a field for the type of the new default serializer.
After that's done, you must create new versions of 'test/resources/data*' because the new TestData instance will no longer be equals the formerly written/serialized one. expected:<53> but was:<54>


Is there some documentation on this?
